cmd/maya-exporter/app/collector/cstor.go

```diff
@@ -0,0 +1,211 @@
+package collector
+
+import (
+	"bytes"
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/types/v1"
+)
+
+const (
+
+	// SocketPath where istgt is listening
+	SocketPath = "/var/run/istgt_ctl_sock"
+	// HeaderPrefix is the prefix comes in the header from cstor.
+	HeaderPrefix = "iSCSI Target Controller version"
+	// EOF separates the strings from response which comes from the
+	// cstor as the collection of metrics.
+	EOF = "\r\n"
+	// Footer is used to verify if all the response has collected.
+	Footer = "OK IOSTATS"
+	// Command is a command that is used to write over wire and get
+	// the iostats from the cstor.
+	Command = "IOSTATS"
+	// BufSize is the size of response from cstor.
+	BufSize = 1024
+)
+
+// cstor implements the Exporter interface. It exposes
+// the metrics of a OpenEBS (cstor) volume.
+type cstor struct {
+	conn  net.Conn
+	stats stats
+}
+
+func Cstor() *cstor {
+	return &cstor{
+		stats: stats{},
+	}
+}
+
+// InitiateConnection tries to initiates the connection with the cstor
+// over unix domain socket. This function can not be unit tested (only
+// negative cases are possible).
+func (c *cstor) InitiateConnection(path string) error {
+	conn, err := net.Dial("unix", path)
+	if err != nil {
+		return err
+	}
+	if conn != nil {
+		c.conn = conn
+		glog.Info("Connection established with istgt")
```

Comment:
Will it be better to try a istgt command & get success before printing this Info?

Reply:
yes